addOns/pscan/src/main/java/org/zaproxy/addon/pscan/PassiveScanApi.java

```diff
@@ -358,14 +331,16 @@ public ApiResponse handleApiView(String name, JSONObject params) throws ApiExcep
 
     private ApiResponseSet<String> getResponseForTask(PassiveScanTask task, String name) {
         Map<String, String> map = new HashMap<>();
-        PassiveScanner scanner = task.getCurrentScanner();
-        map.put("name", scanner == null ? "" : scanner.getName());
-        map.put("url", task.getURI().toString());
-        long time = task.getStartTime();
-        if (time > 0) {
-            time = System.currentTimeMillis() - time;
+        if (task != null) {
+            PassiveScanner scanner = task.getCurrentScanner();
+            map.put("name", scanner == null ? "" : scanner.getName());
+            map.put("url", task.getURI().toString());
+            long time = task.getStartTime();
+            if (time > 0) {
+                time = System.currentTimeMillis() - time;
```

Comment:
It's probably not terribly important but I thought I'd read recently that Instant.now() is preferable

Reply:
Preferable to what specifically?

Reply:
System.currentTimeMillis

Reply:
I understand you are referring to that but not the whole, is that just to get the milliseconds or calculate the duration/elapsed time?

Reply:
It was probably related to calculating duration

Reply:
Okay, given here we are just keeping the milliseconds I don't see much advantage.